src/app/services/solicitud.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Solicitud } from '../models/solicitud.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SolicitudService {
+  private apiUrl = 'http://localhost:4200/solicitudes'; // URL base de la API
+  private solicitudesUrl = 'assets/solicitudes.json'; // Ruta al archivo JSON
+  private solicitudesSubject = new BehaviorSubject<Solicitud[]>([]);
+  solicitudes$ = this.solicitudesSubject.asObservable();
+
+  constructor(private http: HttpClient) {
+    this.cargarSolicitudes();
+  }
+
+  // Método para cargar todas las solicitudes desde el archivo JSON
+  private cargarSolicitudes(): void {
+    this.http.get<Solicitud[]>(this.solicitudesUrl).pipe(
+      map(data => data.map(item => new Solicitud(
+        item.id,
+        item.nombreCompleto,
+        item.email,
+        new Date(item.fechaNacimiento),
+        item.aniosExperiencia,
+        item.puestoSolicitado,
+        item.fechaSolicitud, // Convertir la fecha aquí
+        item.estado
+      )))
+    ).subscribe(solicitudes => this.solicitudesSubject.next(solicitudes));
```

Comment:
Ojo: como norma general nunca hay que suscribirse a un  observable dentro de tu servicio Http, la suscripción la debe hacer quien consuma tu servicio. Al crear una suscripción estas utilizando memoria y si no llega la desuscripción entonces tendras leaks de memoría.

En este caso, para hacer algo "adicional" despues de recibir tus datos y sin suscribirte puedes hacerlo así:

```suggestion
    return this.http.get<Solicitud[]>(this.solicitudesUrl).pipe(
      map(data => data.map(item => new Solicitud(
        item.id,
        item.nombreCompleto,
        item.email,
        new Date(item.fechaNacimiento),
        item.aniosExperiencia,
        item.puestoSolicitado,
        item.fechaSolicitud, // Convertir la fecha aquí
        item.estado
      ))),
      tap((solicitudes) => { // aquí recibirás ya las solicitudes tal y como las has mapeado arriba
          this.solicitudesSubject.next(solicitudes);
      });  // y sin suscribirte, eso lo hará quien llame a tu método
```